ddtrace/utils/formats.py

```diff
@@ -89,3 +93,31 @@ def flatten_dict(d, sep=".", prefix=""):
         if isinstance(d, dict)
         else {prefix: d}
     )
+
+
+def parse_tags_str(tags_str):
+    """Parse a string of tags typically provided via environment variables.
+
+    The expected string is of the form::
+        "key1:value1,key2:value2"
+
+    :param tags_str: A string of the above form to parse tags from.
+    :return: A dict containing the tags that were parsed.
+    """
+    parsed_tags = {}
+    if not tags_str:
+        return parsed_tags
+
+    for tag in tags_str.split(","):
+        try:
+            key, value = tag.split(":", 1)
```

Comment:
you can also use `key, _, value = tag.partition(":")` will basically promise that it won't fail to parse.